### PR TITLE
Ch8 tweaks

### LINKED ISSUE #281: Ch 8 capitalization issues
```
• (list of list -> list) with tz.concat
• don’t get caught out:
— iterators get consumed. So if you make a generator object and do
some processing on it, and then a later step fails, you need to recreate
the generator. The original is already gone.
— iterators are lazy. You need to force evaluation sometimes.
```

### PR DESCRIPTION
**Pull request by jni**
Fixes #281 and a whole bunch of other things.

---
This change is [Reviewable](https://reviewable.io/reviews/elegant-scipy/elegant-scipy/282)

### CODE REVIEW

File: markdown/ch8.markdown [hunk 1]
```diff
@@ -30,16 +30,12 @@ print(sum_of_means)
...
-includes most of this book!)
-
-But in streaming programs, a function processes *some* of the data, returns the
+In streaming programs, a function processes *some* of the input data, returns the
 processed chunk, then, while downstream functions are dealing with that chunk,
 the function receives a bit more, and so on...  All these things are going on
 at the same time!  How can one keep them straight?
 
-We too found this difficult to do, until we found the `toolz` library.
+We too found this difficult, until we found the `toolz` library.
```
Comment (stefanv) [inline markdown/ch8.markdown]: found...found.  How about "We too found this difficult, until discovering the `tools` library."

File: markdown/ch8.markdown [hunk 2]
```diff
@@ -233,8 +229,8 @@ in the fruit-fly genome:
 ```python
 %%timeit -r 1 -n 1
 dm = 'data/dm6.fa'
-model = tz.pipe(dm, genome, c.take(1000000), markov)
-# we use `take` to just run on the first 1000000 bases, to speed things up.
```
Comment (stefanv) [inline markdown/ch8.markdown]: You meant to take 10x more here, right?

File: markdown/ch8.markdown [hunk 3]
```diff
@@ -464,15 +467,15 @@ Now let's leave out the second variable.
 add_curried(2)
 ```
 
-As we expected, it returned a function. Yay!
-Now let's see if we can use that returned function as expected.
+As we expected, it returned a function.
```
Comment (stefanv) [inline markdown/ch8.markdown]: Good!

File: markdown/ch8.markdown (hunk 2 above)
Comment (jni) [inline markdown/ch8.markdown]: Yes

**Comment by jni** [comment]
Here also, f-strings are all single-line.

**Pull request merged**